Framework/Kernel/src/StringTokenizer.cpp

```diff
@@ -0,0 +1,171 @@
+//
+//  StringTokenizer.cpp
+//  Mantid
+//
+//  Created by Hahn, Steven E. on 1/31/16.
+//
+//
+
+#include "MantidKernel/StringTokenizer.h"
+#include <algorithm>
+#include <functional>
+#include <cassert>
+#include <iterator> //cbegin,cend
+
+namespace {
+
+// implement our own trim function to avoid the locale overhead in boost::trim.
+
+// trim from start
+void ltrim(std::string &s) {
+  s.erase(s.begin(), std::find_if_not(s.begin(), s.end(), ::isspace));
+}
+
+// trim from end
+void rtrim(std::string &s) {
+  s.erase(std::find_if_not(s.rbegin(), s.rend(), ::isspace).base(), s.end());
+}
+
+// trim from both ends
+void trim(std::string &s) {
+  ltrim(s);
+  rtrim(s);
+}
+
+// If the final character is a separator, we need to add an empty string to
+// tokens.
+void emptyFinalToken(const std::string &str, const std::string &delims,
+                     std::vector<std::string> &tokens) {
+  const auto pos = std::find_first_of(str.end() - 1, str.end(), delims.begin(),
+                                      delims.end());
+  if (pos != str.end()) {
+    tokens.emplace_back();
+  }
+}
+
+// generic tokenizer using std::find_first_of modelled after
+// http://tcbrindle.github.io/posts/a-quicker-study-on-tokenising/
+// MIT licensed.
+template <class InputIt, class ForwardIt, class BinOp>
+void for_each_token(InputIt first, InputIt last, ForwardIt s_first,
+                    ForwardIt s_last, BinOp binary_op) {
+  while (first != last) {
+    const auto pos = std::find_first_of(first, last, s_first, s_last);
+    binary_op(first, pos);
+    if (pos == last)
+      break;
+    first = std::next(pos);
+  }
+}
+
+// split keeping whitespace and empty tokens.
+std::vector<std::string> split0(const std::string &str,
+                                const std::string &delims) {
+  std::vector<std::string> output;
+  for_each_token(str.cbegin(), str.cend(), delims.cbegin(), delims.cend(),
+                 [&output](std::string::const_iterator first,
+                           std::string::const_iterator second) {
+                   output.emplace_back(first, second);
+                 });
+  return output;
+}
+
+// split keeping whitespace and ignoring empty tokens.
+std::vector<std::string> split1(const std::string &str,
```

Comment:
Maybe the naming could be more descriptive on these three functions. They also look very similar, could this be a potential opportunity to refactor further at a later point?